odoo/models.py

```diff
@@ -4317,7 +4317,8 @@ def sudo(self, user=SUPERUSER_ID):
             The returned recordset has the same prefetch object as ``self``.
 
         """
-        return self.with_env(self.env(user=user))
+        context = dict(self._context, sudo_user=self.env.user)
```

Comment:
frozendict?

Reply:
I copy-pasted it from with_context method